dsl_parser/framework/parser.py

```diff
@@ -439,6 +439,22 @@ def _calculate_element_graph(self):
                         self.element_graph.add_edge(element, dep)
                     continue
 
+                if predicates == [sibling_predicate]:
+                    # These appear to be generally in order, which means
+                    # that we can cut parsing time by a massive amount for
+                    # plugins which have a lot of DSL elements.
+                    # If we don't do this, our time complexity is n**2 as
+                    # we compare all 'default' elements to all 'type'
+                    # elements (for example), when all we care about is if
+                    # they are siblings.
+                    for dependency in dependencies:
+                        for element in dependency.parent().children():
+                            if element in _elements:
```

Comment:
`set() & set()` would work well here, although it doesn't really speed it up by much.
But it seems that in general `self.element_type_to_elements` would be better as a dict of sets, rather than a dict of lists as it currently is.

Reply:
Discussed via slack- verdict: Not worth the extra time/investigation today.